### DISCUSSION TITLE
Is there some best practice to automatically test setups?

### DISCUSSION THREAD

**Discussion by ingomohr** [eclipse-oomph/oomph#19] [Q&A] [answered]
Dear all,

we maintain a number of setups for custom IDE configurations, project import tasks and preference tasks in our own Oomph Git repository.
We are testing our changes in those `*.setup` files manually, but this is both expensive and error-prone.

Is there a way of running automated tests to have proof that the setups will work as expected in the installer?

Thank you very much in advance for your thoughts!

Kind regards
Ingo

**Comment by merks** [comment, marked as answer]
That's a very good question for which I have a very poor answer.  We don't have such tests either, but there would certainly be value in such things.  😱 

One general problem of course is that various things are prompted so one needs to provide such values.   One place where we do sort of run such tests is here where we are generating setups to test marketplace listings:

[org.eclipse.oomph.setup.internal.installer.MarketplaceCatalogGenerator.test()](https://github.com/eclipse-oomph/oomph/blob/289247fa86dda9fc7a2ebf45e1f4d3d455e60c6c/plugins/org.eclipse.oomph.setup.installer/src/org/eclipse/oomph/setup/internal/installer/MarketplaceCatalogGenerator.java#L942)

I've not looked in detail at this project:

https://github.com/a-langer/eclipse-oomph-console

But I assume that in principle, this could be used to drive tests.  I would be very much interest in your experience with that if you try it.

**Reply by ingomohr** [reply]
Thank you @merks for your quick response and the links! I'll have a look and hope I can give you an update myself soon. :)

**Reply by szarnekow** [reply]
@ingomohr Did you figure a mechanism to automatically test setups?

**Reply by ingomohr** [reply]
Hi @szarnekow - no, sorry. I had other problems to tackle in the meantime. I'm still interested in this topic, though. Just didn't find the time.
